Documentation/ceph-upgrade.md

```diff
@@ -48,116 +49,33 @@ With this upgrade guide, there are a few notes to consider:
-```
-
-3. List all volumesnapshotclasses created
-
-```bash
-kubectl get volumesnapshotclass
-NAME                      AGE
-csi-rbdplugin-snapclass   86s
-```
-
-4. Delete all volumesnapshotclasses
-
-```bash
-kubectl delete volumesnapshotclass csi-rbdplugin-snapclass
-volumesnapshotclass.snapshot.storage.k8s.io "csi-rbdplugin-snapclass" deleted
-```
-
-*Note:* The underlying snapshots on the storage system will be deleted by ceph-csi
-
-#### Delete the Alpha CRDs
-
-As we are updating the snapshot resources from `Alpha` to `Beta` we need to
-delete the old `alphav1` snapshot CRD created by external-snapshotter sidecar container
-
-Check if we have any `v1alpha1` CRD created in our kubernetes cluster
-
-```console
-kubectl get crd volumesnapshotclasses.snapshot.storage.k8s.io -o yaml |grep v1alpha1
-  - name: v1alpha1
-  - v1alpha1
-kubectl get crd volumesnapshotcontents.snapshot.storage.k8s.io -o yaml |grep v1alpha1
-  - name: v1alpha1
-  - v1alpha1
-kubectl get crd volumesnapshots.snapshot.storage.k8s.io -o yaml |grep v1alpha1
-  - name: v1alpha1
-  - v1alpha1
-```
-
-As we have `v1alpha1` CRD created in our kubernetes cluster, we need to delete
-the Alpha CRD
-
-```console
-kubectl delete crd volumesnapshotclasses.snapshot.storage.k8s.io volumesnapshotcontents.snapshot.storage.k8s.io volumesnapshots.snapshot.storage.k8s.io
-
-customresourcedefinition.apiextensions.k8s.io "volumesnapshotclasses.snapshot.storage.k8s.io" deleted
-customresourcedefinition.apiextensions.k8s.io "volumesnapshotcontents.snapshot.storage.k8s.io" deleted
-customresourcedefinition.apiextensions.k8s.io "volumesnapshots.snapshot.storage.k8s.io" deleted
-```
-
-Finally, if you desire to use the beta snapshots, check that the [prerequisites](ceph-csi-snapshot.md#prerequisites) are met.
-
+TODO: are there breaking changes?
 
 ## Upgrading the Rook-Ceph Operator
 
 ## Patch Release Upgrades
 
 Unless otherwise noted due to extenuating requirements, upgrades from one patch release of Rook to
-another are as simple as updating the image of the Rook operator. For example, when Rook v1.4.1 is
-released, the process of updating from v1.4.0 is as simple as running the following:
+another are as simple as updating the image of the Rook operator. For example, when Rook v1.5.1 is
+released, the process of updating from v1.5.0 is as simple as running the following:
 
 ```console
-kubectl -n rook-ceph set image deploy/rook-ceph-operator rook-ceph-operator=rook/ceph:v1.4.1
+kubectl -n rook-ceph set image deploy/rook-ceph-operator rook-ceph-operator=rook/ceph:v1.5.1
 ```
 
 ## Helm Upgrades
 
 If you have installed Rook via the Helm chart, Helm will handle some details of the upgrade for you.
-In particular, Helm will handle updating the RBAC and trigger the operator update and restart.
+In particular, Helm will handle updating the RBAC and CRDs and will trigger the operator update.
+Helm will **not** update the Ceph cluster itself.
```

Comment:
From the user's perspective, there won't be any steps to upgrade after updating the helm chart, right? It is true the helm chart doesn't update the CRs, but the CRs don't need to be updated during an upgrade. Can we remove this line?

Reply:
Thinking about this .... I don't think the user needs to follow any of the upgrade steps for Rook if using Helm. But they might need to refer to Ceph upgrade steps. I think that is all good to explain, yes?

Reply:
They should still verify the upgrade, right? (steps 4 and 5)

Reply:
Helm users should still read the upgrade guide and it will help them verify the upgrade, like you mentioned with steps 4 and 5. But I think the statement is confusing: "Helm will not update the Ceph cluster itself". If they want to update Ceph, they can just refer to the other section on updating Ceph. But that's optional anyway whether it's a helm install or otherwise.

Reply:
I see. I think we can remove this whole "Helm Upgrades" section